packages/headless/src/modules/order/services/review.service.ts

```diff
@@ -1,91 +1,174 @@
+        errors: null,
+        code: HttpStatus.BAD_REQUEST,
+      };
+
+    const { products } = order;
+    const productExists = products.find(
+      (product) => productId === product.productId,
+    );
+
+    if (!productExists)
+      return {
+        error: CreateReviewErrorMessage.INVALID_PRODUCT_ID,
+        errors: null,
+        code: HttpStatus.BAD_REQUEST,
+      };
+
+    //userId validity done from guards
+    //commented by should also be handled from guards
+
+    //if rating has then proceed to add it to product
+    if (rating) {
+      const productRating = await this.orderRepository.addProductRating(
+        productId,
+        rating,
+      );
+      if (!productRating)
+        return {
+          error: CreateReviewErrorMessage.CANNOT_ADD_RATING,
+          errors: null,
+          code: HttpStatus.BAD_REQUEST,
+        };
     }
+    //check if images are not more than 5
+    //if greater send error
+    if (image && image.length > 5)
+      return {
+        error: CreateReviewErrorMessage.CANNOT_UPLOAD_MORE_THAN_5_PHOTOS,
+        errors: null,
+        code: HttpStatus.BAD_REQUEST,
+      };
+
+    const review = await this.orderRepository.createReview(request);
+    if (!review)
+      return {
+        error: CreateReviewErrorMessage.CANNOT_CREATE_REVIEW,
+        errors: null,
+        code: HttpStatus.INTERNAL_SERVER_ERROR,
+      };
 
-    async getProductReview(productId: string, skip: number = 0, limit: number = 0): Promise<ProductReviewResponse>{
-        const reviews = await this.orderRepository.findReview({ productId }, skip, limit);
+    return { code: 200, data: review };
+  }
 
-        if(!reviews)
-            return {
-                error: ProductReviewErrorMessage.CANNOT_FIND_REVIEW,
-                errors: null,
-                code: HttpStatus.INTERNAL_SERVER_ERROR
-            };
+  async getProductReview(
+    productId: string,
+    skip: number = 0,
+    limit: number = 0,
+  ): Promise<ProductReviewResponse> {
+    const reviews = await this.orderRepository.findReview(
+      { productId },
+      skip,
+      limit,
+    );
 
-        const response = {
-            productId,
-            reviews
-        }
+    if (!reviews)
+      return {
+        error: ProductReviewErrorMessage.CANNOT_FIND_REVIEW,
+        errors: null,
+        code: HttpStatus.INTERNAL_SERVER_ERROR,
+      };
 
-        return { code: 200, data: response };
+    const response = {
+      productId,
+      reviews,
+    };
+
+    return { code: 200, data: response };
+  }
+
+  async createReply(request: ICreateReply): Promise<ICreateReplyResponse> {
+    const { reviewId } = request;
+    const review = await this.orderRepository.findReview({ id: reviewId });
+    if (review.length === 0)
+      return {
+        error: CreateReplyErrorMessage.INVALID_REVIEW_ID,
+        errors: null,
+        code: HttpStatus.INTERNAL_SERVER_ERROR,
+      };
+
+    //add the user id from the guards
```

Comment:
in this type of case, start the comment with TODO: so that you can easily find those incomplete tasks in future